File: design-patterns/ts/strategy/transport/plane.ts
// plane.ts

import Transport from "./transport";


class Plane implements Transport {
    start(): void {
        console.log("Plane started");
    }

    stop(): void {
        console.log("Plane stopped");
    }

    repair(): void {
        console.log("Plane repair");
    }

    getInfo(): void {
        console.log("Transport type: Plane");
        console.log("Number of wheels: 3");
        console.log("Average Weight: 50,000 Pounds");
    }

    operate(): void {
        console.log("Flying plane ............");
    }
}

export default Plane;

File: design-patterns/go/strategy/transport/plane.go
// plane.go

package main

import "fmt"

type Plane struct {
}

func NewPlane() (plane *Plane) {
	plane = &Plane{}
	return
}

func (plane *Plane) GetInfo() {
	fmt.Println("Transport type: Plane")
	fmt.Println("Number of wheels: 3")
	fmt.Println("Average Weight: 50,000 Pounds")
}

func (plane *Plane) Operate() {
	fmt.Println("Flying plane ............")
}

func (plane *Plane) Repair() {
	fmt.Println("Plane repair")
}

func (plane *Plane) Start() {
	fmt.Println("Plane started")
}

func (plane *Plane) Stop() {
	fmt.Println("Plane stopped")
}
